tests/unit/s2n_client_psk_extension_test.c

```diff
@@ -0,0 +1,654 @@
+
+            struct s2n_stuffer wire_identities_in;
+            EXPECT_SUCCESS(s2n_stuffer_alloc(&wire_identities_in, sizeof(wire_identites)));
+            EXPECT_SUCCESS(s2n_stuffer_write_bytes(&wire_identities_in, wire_identites, sizeof(wire_identites)));
+
+            EXPECT_OK(s2n_client_psk_recv_identity_list(conn, &wire_identities_in));
+            EXPECT_NULL(conn->psk_params.chosen_psk);
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+            EXPECT_SUCCESS(s2n_stuffer_free(&wire_identities_in));
+        }
+
+        /* Receive a list with an immediate match */
+        {
+            struct s2n_connection *conn;
+            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
+
+            struct s2n_stuffer wire_identities_in;
+            EXPECT_SUCCESS(s2n_stuffer_alloc(&wire_identities_in, sizeof(single_wire_identity)));
+            EXPECT_SUCCESS(s2n_stuffer_write_bytes(&wire_identities_in, single_wire_identity, sizeof(single_wire_identity)));
+
+            struct s2n_psk *match_psk;
+            EXPECT_OK(s2n_array_pushback(&conn->psk_params.psk_list, (void**) &match_psk));
+            EXPECT_SUCCESS(s2n_psk_init(match_psk, S2N_PSK_TYPE_EXTERNAL));
+            EXPECT_SUCCESS(s2n_psk_new_identity(match_psk, test_bytes_data, sizeof(test_bytes_data)));
+
+            EXPECT_OK(s2n_client_psk_recv_identity_list(conn, &wire_identities_in));
+            EXPECT_EQUAL(conn->psk_params.chosen_psk, match_psk);
+            EXPECT_EQUAL(conn->psk_params.chosen_psk_wire_index, 0);
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+            EXPECT_SUCCESS(s2n_stuffer_free(&wire_identities_in));
+        }
+
+        /* Receive a list with a match later in the list */
+        {
+            struct s2n_connection *conn;
+            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
+
+            struct s2n_stuffer wire_identities_in;
+            EXPECT_SUCCESS(s2n_stuffer_alloc(&wire_identities_in, sizeof(wire_identites)));
+            EXPECT_SUCCESS(s2n_stuffer_write_bytes(&wire_identities_in, wire_identites, sizeof(wire_identites)));
+
+            struct s2n_psk *match_psk;
+            EXPECT_OK(s2n_array_pushback(&conn->psk_params.psk_list, (void**) &match_psk));
+            EXPECT_SUCCESS(s2n_psk_init(match_psk, S2N_PSK_TYPE_EXTERNAL));
+            EXPECT_SUCCESS(s2n_psk_new_identity(match_psk, test_bytes_data, sizeof(test_bytes_data)));
+
+            EXPECT_OK(s2n_client_psk_recv_identity_list(conn, &wire_identities_in));
+            EXPECT_EQUAL(conn->psk_params.chosen_psk, match_psk);
+            EXPECT_EQUAL(conn->psk_params.chosen_psk_wire_index, wire_identities_match_index);
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+            EXPECT_SUCCESS(s2n_stuffer_free(&wire_identities_in));
+        }
+    }
+
+    /* Test: s2n_client_psk_recv_binder_list */
+    {
+        const uint8_t filler_binders[] = {
+                0x00, /* binder size */
+
```

Comment:
Is this newline conveying something?

Reply:
It made more sense when I had multiple binders in the list :) I'll remove it now.

Reply:
OH wait that's why it's there. I left newlines between the binders in the list. One of the binders is just 0-length, so has no value.